src/overlays/actors/ovl_En_Jg/z_en_jg.c

```diff
@@ -74,70 +188,904 @@ static DamageTable D_80B75858 = {
+    { &gGoronElderCradleAnim, 1.0f, 0, -1, 0, 0 },
+};
+
+static Vec3f sSfxPos = { 0.0f, 0.0f, 0.0f };
+
+static Vec3f sFocusOffset = { 0.0f, 0.0f, 0.0f };
+
+static Vec3f sBreathPosOffset = { 1000.0f, -500.0f, 0.0f };
+
+static Vec3f sBreathVelOffset = { 0.0f, 0.0f, 0.75f };
+
+static Vec3f sBreathAccelOffset = { 0.0f, 0.0f, -0.070000000298f };
+
+/**
+ * When the elder and the Gorons in the shrine are doing a cheer
+ * for Darmani, the camera will focus on specific Gorons for
+ * certain parts of the cheer. This function is responsible for
+ * returning a pointer to the specified Goron so the camera
+ * can focus on them.
+ */
+Actor* EnJg_GetShrineGoronToFocusOn(GlobalContext* globalCtx, u8 focusedShrineGoronParam) {
+    Actor* actorIterator = globalCtx->actorCtx.actorLists[ACTORCAT_NPC].first;
+
+    while (actorIterator != NULL) {
+        if ((actorIterator->id == ACTOR_EN_S_GORO) &&
+            (EN_S_GORO_GET_PARAM_F(actorIterator) == focusedShrineGoronParam)) {
+            return actorIterator;
+        }
+
+        actorIterator = actorIterator->next;
+    }
+
+    return NULL;
+}
+
+void EnJg_UpdateCollision(EnJg* this, GlobalContext* globalCtx) {
+    this->collider.dim.pos.x = this->actor.world.pos.x;
+    this->collider.dim.pos.y = this->actor.world.pos.y;
+    this->collider.dim.pos.z = this->actor.world.pos.z;
+
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 0.0f, 30.0f, 30.0f, 7);
+}
+
+s16 EnJg_GetWalkingYRotation(Path* path, s32 pointIndex, Vec3f* pos, f32* distSQ) {
+    Vec3s* points;
+    f32 diffX;
+    f32 diffZ;
+
+    if (path != NULL) {
+        points = Lib_SegmentedToVirtual(path->points);
+        points = &points[pointIndex];
+        diffX = points[0].x - pos->x;
+        diffZ = points[0].z - pos->z;
+    } else {
+        diffX = 0.0f;
+        diffZ = 0.0f;
+    }
+
+    *distSQ = SQ(diffX) + SQ(diffZ);
+
+    return RADF_TO_BINANG(Math_Acot2F(diffZ, diffX));
+}
+
+s32 EnJg_ReachedPoint(EnJg* this, Path* path, s32 pointIndex) {
+    Vec3s* points = Lib_SegmentedToVirtual(path->points);
+    s32 pathCount = path->count;
+    s32 currentPoint = pointIndex;
+    s32 reached = false;
+    f32 diffX;
+    f32 diffZ;
+    f32 px;
+    f32 pz;
+    f32 d;
+    Vec3f point;
+
+    Math_Vec3s_ToVec3f(&point, &points[pointIndex]);
+    if (currentPoint == 0) {
+        diffX = points[1].x - points[0].x;
+        diffZ = points[1].z - points[0].z;
+    } else {
+        if (currentPoint == (pathCount - 1)) {
+            diffX = points[pathCount - 1].x - points[pathCount - 2].x;
+            diffZ = points[pathCount - 1].z - points[pathCount - 2].z;
+        } else {
+            diffX = points[currentPoint + 1].x - points[currentPoint - 1].x;
+            diffZ = points[currentPoint + 1].z - points[currentPoint - 1].z;
+        }
+    }
+
```

Comment:
```suggestion
    if (currentPoint == 0) {
        diffX = points[1].x - points[0].x;
        diffZ = points[1].z - points[0].z;
    } else if (currentPoint == (pathCount - 1)) {
        diffX = points[pathCount - 1].x - points[pathCount - 2].x;
        diffZ = points[pathCount - 1].z - points[pathCount - 2].z;
    } else {
        diffX = points[currentPoint + 1].x - points[currentPoint - 1].x;
        diffZ = points[currentPoint + 1].z - points[currentPoint - 1].z;
    }

```
I'd prefer an if..else if.. else